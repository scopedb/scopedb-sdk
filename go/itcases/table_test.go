/*
 * Copyright 2024 ScopeDB, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package itcases

import (
	"context"
	"fmt"
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/stretchr/testify/require"
)

func TestTableSchema(t *testing.T) {
	c := NewClient(t)
	defer c.Close()

	ctx := context.Background()
	tbl := c.Table(RandomName(t))
	_, err := c.Statement(fmt.Sprintf(`
		CREATE TABLE %s (
			i INT,
			u UINT,
			f FLOAT,
			s STRING,
			b BOOLEAN,
			ts TIMESTAMP,
			var any,
		)
	`, tbl.Identifier())).Execute(ctx)
	require.NoError(t, err)
	defer func() {
		require.NoError(t, tbl.Drop(ctx))
	}()

	schema, err := tbl.TableSchema(ctx)
	require.NoError(t, err)
	snaps.MatchSnapshot(t, schema)
}
