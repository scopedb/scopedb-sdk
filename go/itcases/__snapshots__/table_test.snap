
[TestTableSchema - 1]
scopedb.Schema{
    &scopedb.FieldSchema{Name:"i", Type:"int"},
    &scopedb.FieldSchema{Name:"u", Type:"uint"},
    &scopedb.FieldSchema{Name:"f", Type:"float"},
    &scopedb.FieldSchema{Name:"s", Type:"string"},
    &scopedb.FieldSchema{Name:"b", Type:"boolean"},
    &scopedb.FieldSchema{Name:"ts", Type:"timestamp"},
    &scopedb.FieldSchema{Name:"var", Type:"any"},
}
---
